openmeter/event/metadata/entity.go

```diff
@@ -0,0 +1,70 @@
+package metadata
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+// EntityMutationType is the type of mutation that occurred on an entity
+type EntityMutationType string
+
+const (
+	EntityMutationTypeCreate EntityMutationType = "create"
+	EntityMutationTypeUpdate EntityMutationType = "update"
+	EntityMutationTypeDelete EntityMutationType = "delete"
+)
+
+// EventEntityMutationPayload is a payload for an entity mutation event
+type EventEntityMutationPayload[T any] struct {
+	Entity       string             `json:"entity"`       // entity type "customer"
```

Comment:
This should be known from the event type. The whole event type/interface exists so that we don't need to deal with this dynamically. 

You just subscribe to the event type and know the payload's type.

Also given the event might be extended with additional fields (such as DeletedAt) I don't see the value of having the type captured here too.